### DISCUSSION TITLE
Custom admin path not working

### DISCUSSION THREAD

**Discussion by Advoskr** [strapi/strapi#6483] [Legacy Topics]
**Describe the bug**
I want to use custom admin route, say /dashboard and proxy it with nginx using subfolders (doesn't matter, unified or not). I follow the steps described in documentation and used nginx config and `server.js` file. Unfortunately, admin page doesn't work (stays blank).

**Steps to reproduce the behavior**

1. Create dockerized strapi setup with docker-compose, containing postgres, nginx and strapi itself.
2. Setup server.js as follows:
```
module.exports = ({ env }) => ({
  host: env('HOST', '0.0.0.0'),
  port: env.int('PORT', 1337),
  url: 'http://'+env('HOST', 'localhost')+'/api',
  admin: {
    url: `/panel`,
  },
});
```
In my case, i'm trying to make everything working on localhost, so host is `localhost`

3. Setup nginx as follows:
```
server {
    # Listen HTTP

    listen 80;
    # Strapi API
    location /api/ {
        rewrite ^/api/(.*)$ /$1 break;
        proxy_pass http://strapi:1337;
        proxy_http_version 1.1;
        proxy_set_header X-Forwarded-Host $host;
        proxy_set_header X-Forwarded-Server $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header Host $http_host;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";
        proxy_pass_request_headers on;
    }

    # Strapi Dashboard
    location /dashboard {
        proxy_pass http://strapi:1337/dashboard;
        proxy_http_version 1.1;
        proxy_set_header X-Forwarded-Host $host;
        proxy_set_header X-Forwarded-Server $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header Host $http_host;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "Upgrade";
        proxy_pass_request_headers on;
    }
}
```
I dont' setup an upstream in nginx, so port is defined explicitly.

**Expected behavior**
api is available on `localhost/api` and admin UI on `localhost\dashboard`

**As-is behaviour**
UI tries to load js-scripts not from /dashboard, but from /admin for some reason. I've tried to make more complicated nginx config (see below), but all i got - rotating spinner (even when all requests were successfully completed).

**Code snippets**
More complicated nginx config:
```
 server {
        listen 80;
        expires $expires;

        access_log   /var/log/nginx.access_log  main;
        rewrite_log on;

        # Strapi API and Admin
        location /api/ {
            rewrite ^/api/(.*)$ /$1 break;
            proxy_pass http://strapi:1337;
            proxy_http_version 1.1;
            proxy_set_header X-Forwarded-Host $host;
            proxy_set_header X-Forwarded-Server $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $scheme;
            proxy_set_header Host $http_host;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "Upgrade";
            proxy_pass_request_headers on;
        }

        # Strapi Admin panel
        location /dashboard {
            proxy_pass http://strapi:1337/dashboard;
            proxy_http_version 1.1;
            proxy_set_header X-Forwarded-Host $host;
            proxy_set_header X-Forwarded-Server $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $scheme;
            proxy_set_header Host $http_host;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "Upgrade";
            proxy_pass_request_headers on;
        }

        location /users-permissions {
            proxy_pass http://strapi:1337;
            proxy_http_version 1.1;
            proxy_set_header X-Forwarded-Host $host;
            proxy_set_header X-Forwarded-Server $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $scheme;
            proxy_set_header Host $http_host;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "Upgrade";
            proxy_pass_request_headers on;
        }

        location /admin {
            proxy_pass http://strapi:1337/dashboard;
            proxy_http_version 1.1;
            proxy_set_header X-Forwarded-Host $host;
            proxy_set_header X-Forwarded-Server $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $scheme;
            proxy_set_header Host $http_host;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "Upgrade";
            proxy_pass_request_headers on;
        }
}
```

**System**
- Node.js version: dockerized strapi 3.0.1
- NPM version: dockerized strapi 3.0.1
- Strapi version: dockerized strapi 3.0.1
- Database: postgres
- Operating system: 

Looks like subfolder-proxied setup, described in documentation is incorrect or out-of-date. 
We are now preparing for launch of our prototype based on strapi and this issue blocks us totally, so i look forward for any help.

PS. Thank you for your awesome product!

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Comment by derrickmehaffy** [comment]
Your config of both Strapi and Nginx are wrong here.

In your example:
```
module.exports = ({ env }) => ({
  host: env('HOST', '0.0.0.0'),
  port: env.int('PORT', 1337),
  url: 'http://'+env('HOST', 'localhost')+'/api',
  admin: {
    url: `/panel`,
  },
});
```

What this is going to give you is:
API: `http://localhost/api`
Admin: `http://localhost/api/panel`

However your nginx config (in the first example) is passing along:
API: `http://localhost/api`
Admin: `http://localhost/dashboard`

As I wrote that documentation (and tested it on the stable release, in a production like environment), the documentation is correct. 

Using: https://strapi.io/documentation/v3.x/deployment/nginx-proxy.html#sub-folder-split would give the following:

API: `http://localhost/api`
Admin: `http://localhost/dashboard`

But it would require something like: https://strapi.io/documentation/v3.x/deployment/nginx-proxy.html#sub-folder-split-strapi-config

You could do this on localhost, just by changing the `https://example.com` to `http://localhost` but you will need direct URLs, not relative. Likewise if you wanted to swap out `dashboard` for `panel` you would need to change both the server.js **and** the nginx config. When these changes are made you also need to rebuild the admin for it to see these changes.

These docs properly explain in detail what each of the options are: https://strapi.io/documentation/v3.x/concepts/configurations.html#server

**Reply by Advoskr** [reply]
Thank you for your reply!

There was a typo in my config, `/panel` was copied from wrong file and my configuration has `/dashboard`. I have now tried again to use sun-folder-split configuration guide from links you provided and still get 404 on js files when loadin admin (/api works as expected). 

**Server.js file:**
```
module.exports = ({ env }) => ({
  host: env('HOST', '0.0.0.0'),
  port: env.int('PORT', 1337),
  url: 'http://'+env('HOST', 'localhost')+'/api',
  admin: {
    url: `http://localhost/dashboard`,
  },
});
```

**nginx configuration:**
```
    server {
        listen 80;
        expires $expires;

        access_log   /var/log/nginx.access_log  main;
        rewrite_log on;

        location / {
            proxy_pass         http://web_app;
            proxy_redirect     off;

            proxy_set_header   Host             $host;
            proxy_set_header   X-Real-IP        $remote_addr;

            client_max_body_size       10m;
            client_body_buffer_size    128k;

            proxy_connect_timeout      70;
            proxy_send_timeout         90;
            proxy_read_timeout         90;

            proxy_buffer_size          4k;
            proxy_buffers              4 32k;
            proxy_busy_buffers_size    64k;
            proxy_temp_file_write_size 64k;

            charset  utf-8;
        }

        # Strapi API and Admin
        location /api/ {
            rewrite ^/api/(.*)$ /$1 break;
            proxy_pass http://strapi:1337;
            proxy_http_version 1.1;
            proxy_set_header X-Forwarded-Host $host;
            proxy_set_header X-Forwarded-Server $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $scheme;
            proxy_set_header Host $http_host;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "Upgrade";
            proxy_pass_request_headers on;
        }

        # Strapi Admin panel
        location /dashboard {
            proxy_pass http://strapi:1337/dashboard;
            proxy_http_version 1.1;
            proxy_set_header X-Forwarded-Host $host;
            proxy_set_header X-Forwarded-Server $host;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Proto $scheme;
            proxy_set_header Host $http_host;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "Upgrade";
            proxy_pass_request_headers on;
        }
    }
```

I don't know why, but admin page is trying to load js files from
`http://localhost/admin/runtime~main.2be3c7d7.js`

**Comment by pcP1r4t3** [comment]
Hello, the same issue as @Advoskr mentionned !

**Comment by baytelman** [comment]
Hello! After fighting with the documentations, env config, etc, I saw in this thread that the appropriate config option to change the admin path is `url`, and not `path` as stated in the docs.

Should I open a PR to update this?
https://strapi.io/documentation/v3.x/admin-panel/customization.html#change-access-url

**Reply by Advoskr** [reply]
@baytelman Have you succeed in configuring strapi with sub-folder proxing in nginx? can you share your configuration, please?

**Reply by baytelman** [reply]
I ended up following another section of the config, so it did work-ish. I now ran into another issue: 
https://github.com/strapi/strapi/issues/6535

**Comment by sunnysonx** [comment]
Same here.
In my case "path" doesn't change anything. Admin still accesssible on /admin.
And the value from path gives 404 when trying to access it.

**Reply by hamidzare** [reply]
you must use url instead of path:
module.exports = ({ env }) => ({
  host: env("HOST", "0.0.0.0"),
  port: env.int("PORT", 1337),
  admin: {
    url: "/dashboard",
  },
});

**Comment by pcP1r4t3** [comment]
Same here, the sub path works fine but the admin panel becomes inaccessible and i get a blank screen.
